### PR TITLE
Moving 6POC to 6element

### PR DESCRIPTION
**Pull request by SergeBredin**

### CODE REVIEW

File: core/clients/6poc/back/report.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      if ( err !== null ){ 
+          console.log(file, '', err);
+          reject(err);
+      }
+      else
+      {
+        var doc = JSON.parse(data);
+        var objects = [];
+        for (var key in doc) {
+          	if (doc.hasOwnProperty(key)) {
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: Not a big deal, but to this pattern, prefer this one:
````js
Object.keys(doc).forEach(function(key){...})
````

File: core/clients/6poc/back/report.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          			lat <= request.square.maxLat &&
+          			request.square.minLon <= lon &&
+          			lon <= request.square.maxLon){
+          			var match = false;
+				      for (var category in object.properties.objects) {
+				          if (object.properties.objects.hasOwnProperty(category)) {
+				            if(object.properties.objects[category] === 1){
+				            	if(request.categories.indexOf(category) !== -1){
+				            		match = true;
+				            		break;
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: Same here. Prefer more explicit forms like [.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).

File: core/clients/6poc/back/report.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+var bodyParser	= require('body-parser');
+var express 	= require('express');
+var app 		= express();
+var fs = require('fs');
+var path = require('path');
+var server 	= require('http').createServer(app);
+
+app.use(bodyParser.json({limit: '1000mb'}));
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: Out of curiosity, why was this needed?

File: core/clients/6poc/back/report.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	  "properties": {
+	    "marker-color": "#0f0"
+	  },
+	  "geometry": {
+	    "type": "Point",
+	    "coordinates": [data.geoloc.lat, data.geoloc.lon]
+	  }
+	};
+	
+	var distance = Math.sqrt(Math.pow(data.radius,2)*2);
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: nit: now, there is [Math.hypot](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot) for this.

Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: ... is this `data.radius * Math.sqrt(2)` ?

Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: or even `data.radius * Math.SQRT2`

File: core/clients/6poc/back/report.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+	var results = [];
+	Promise.all(files
+	.filter(function(file){
+	  return (path.extname(file) === '.json'); 
+	})
+	.map(function(file){
+		return FileToObjects(dir, file, result)
+	  	.then(function(objects){
+			Array.prototype.push.apply(result.objects, objects);
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: nice :-)

File: core/clients/6poc/back/report.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	  	})
+	}))
+	.then(function(){
+		res.setHeader('Content-Type', 'application/json');
+		res.send(JSON.stringify(result));
+	});
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: Add a .catch here otherwise, errors go unnoticed

File: core/clients/6poc/package.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  "bugs": {
+    "url": "https://github.com/anthill/6element/issues"
+  },
+  "homepage": "https://github.com/anthill/6element",
+  "dependencies": {
+    "body-parser": "^1.13.2",
+    "bootstrap": "^3.3.5",
+    "browserify": "^10.2.4",
+    "express": "^4.13.1",
+    "http": "0.0.0",
```
Comment (DavidBruant) [inline core/clients/6poc/package.json]: ?

File: core/clients/6poc/web/css/main.css [hunk 8]
```diff
@@ -0,0 +1,217 @@
+body{
+	background-color: Black; /*WhiteSmoke*/
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: does this work? oh CSS... try to keep things lowercase

File: core/clients/6poc/web/css/main.css [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    height: 100%;
+    width: 100%;
+    font-family: Lato, sans-serif;
+}
+
+a { text-decoration: none !important; }
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: why the `!important`?

File: core/clients/6poc/web/css/main.css [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+button a{
+	color: white;
+	text-decoration: none !important;
+}
+
+address{
+	margin: 0px;
+}
+
+.star { color: gold; }
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: :+1:

File: core/clients/6poc/web/css/main.css [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+#landingPage{
+	height: 682px;
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: Let's avoid px altogether and use ems. Maybe 40ems here?

File: core/clients/6poc/web/css/main.css [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+    -moz-border-radius: 6px;
+    border-radius: 6px;}
+
+#what .chzn-container-multi .chzn-choices {
+	line-height: 50px !important;
+    overflow-y: auto;
+}*/
+
+#landingPage #searchBar{
+	width: 800px;
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: px => em

File: core/clients/6poc/web/css/main.css [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+#resultsPage #searchBar{
+	width: 100%;
+}
+
+#searchBar{
+	margin: 0 auto;
+	width: 800px;
+  	padding: 5px;
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: px => em

File: core/clients/6poc/web/css/main.css [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+  	background-color: rgba(0,0,0,0.5);
+	vertical-align: middle;
+	color: white;
+}
+
+
+#groupCondition{
+	margin-left: 5px;
+	margin-right: 5px;
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: px => em

File: core/clients/6poc/web/css/main.css [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+	font-size: 16px;
+}
+
+#btnSearch {
+	margin-left: 5px;
+	margin-right: 5px;	
+}
+
+#resultsPage{
+	height: 70px;
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: px => em here and everywhere above

File: core/clients/6poc/web/css/main.css [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+#resultsPage{
+	height: 70px; 
+	width: 100%; 
+	display: block;
+	position: relative;
+	text-align: center;
+	vertical-align: middle;
+}
+
+#resultsPage::after{
```
Comment (DavidBruant) [inline core/clients/6poc/web/css/main.css]: Is this a copy/paste of `#landingPage::after`? Make a class ;-)

File: core/clients/6poc/web/index.html [hunk 17]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
```
Comment (DavidBruant) [inline core/clients/6poc/web/index.html]: useless now.

File: core/clients/6poc/web/views/mainView.jsx [hunk 18]
```diff
@@ -0,0 +1,193 @@
...
+      'waste',
+      'waste_oil',
+      'white_goods',
+      'wood']
+    return {result: null, mode: 0, condition: 0, radius: 50, availableCategories: availableCategories, categories: [], searchCpt: 0};
+  },
+  componentDidMount: function() {
+    var self = this;
+
+    var where = React.findDOMNode(this.refs.where);
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mainView.jsx]: Not very react-y, but probably necessary for the interaction with google map autocomplete thing?

File: core/clients/6poc/web/views/mainView.jsx [hunk 19]
```diff
@@ -0,0 +1,193 @@
...
+    google.maps.event.addDomListener(window, 'load', function() {
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mainView.jsx]: what is this for?

File: core/clients/6poc/web/views/mainView.jsx [hunk 20]
```diff
@@ -0,0 +1,193 @@
...
+      var value = options[i].value;
+      categories.push(value);
+    }
+    var data = {
+      'placeName': this.state.placeName,
+      'radius': this.state.radius,
+      'categories': categories,
+      'geoloc': this.state.geoloc
+    };
+    $.ajax({
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mainView.jsx]: At least, this belongs to its own file, so concerns of fetching data and displaying them are separated (the scheduling via `launchSearch` may stay here for now)

File: core/clients/6poc/web/views/mainView.jsx [hunk 21]
```diff
@@ -0,0 +1,193 @@
...
+  },
+  launchSearch: function(){
+    var self = this;
+
+    var options =  React.findDOMNode(this.refs.what).options;
+    var categories = [];
+    for(var i=0; i<options.length; ++i){
+      if(options[i].selected == false) continue;
+      var value = options[i].value;
+      categories.push(value);
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mainView.jsx]: This is `.map`... but I guess `options` is not an array, but some HTMLCollection/NodeList. The trick in ES6 is probably to do `Array.from(options).map()`

File: core/clients/6poc/back/report.js [hunk 22]
```diff
@@ -0,0 +1,16 @@
+var search = require('./search.js');
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: use strict !

File: core/clients/6poc/back/report.js [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+var express 	= require('express');
+var app 		= express();
+var server 	= require('http').createServer(app);
+
+app.use(bodyParser.json({limit: '1000mb'}));
+app.use(bodyParser.urlencoded({extended: true}))
+
+app.use('/', express.static(__dirname + '/..' + '/web'));
+
+app.post('/search', search);
```
Comment (DavidBruant) [inline core/clients/6poc/back/report.js]: try to keep route functions inline.

File: core/clients/6poc/package.json (hunk 7 above)
Comment (DavidBruant) [inline core/clients/6poc/package.json]: ??

File: core/clients/6poc/web/views/mapView.jsx [hunk 24]
```diff
@@ -0,0 +1,361 @@
...
+    this.props.select(index);
+  },
+  sort: function(sort){
+    this.setState({sort: sort, tab: 0});
+  },
+  displayFilter: function(display){
+    this.setState({filterDisplayed: display});
+  },
+  postAdvert: function(){
+    alert("TODO Bro!");
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mapView.jsx]: Prefer `throw 'TODO'` or `console.log('TODO')`

File: core/clients/6poc/web/views/mapView.jsx [hunk 25]
```diff
@@ -0,0 +1,361 @@
...
+      var list = [];
+      for (var key in object.properties.objects) {
+          if (object.properties.objects.hasOwnProperty(key)) {
+            if(object.properties.objects[key] === 1)
+              list.push(key);
+        }
+      }
+      var distance = "";
+      if(object.distance > 1){
+        distance = (Math.round(object.distance * 100) / 100).toString() + " Km";
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mapView.jsx]: `object.distance.toFixed(2)`

File: core/clients/6poc/web/views/mapView.jsx [hunk 26]
```diff
@@ -0,0 +1,361 @@
...
+    .slice(0,30)
+    .map(function(object, index){
+
+      var phoneJSX = "";
+      if(object.properties.phone){
+        phoneJSX =  <div><abbr title="phone">T:</abbr> {object.properties.phone}<br/></div>
+      } 
+      var list = [];
+      for (var key in object.properties.objects) {
+          if (object.properties.objects.hasOwnProperty(key)) {
```
Comment (DavidBruant) [inline core/clients/6poc/web/views/mapView.jsx]: Object.keys().map

**Comment by SergeBredin** [comment]
Validation from David
